### PR TITLE
unit tests rocm

### PR DESCRIPTION
**Pull request by gurpreet-dhami**
- Added marker "@pytest.mark.failing_on_rocm" for all failing tests on rocm.

[test_report_pytests.csv](https://github.com/user-attachments/files/17624104/test_report_pytests.csv)
[unit_test_logs.txt](https://github.com/user-attachments/files/17624105/unit_test_logs.txt)

------------------------- CSV report: test_report.csv --------------------------
**305 passed, 18 skipped, 270 deselected, 955 warnings in 160.58s (0:02:40) ===**

- **Total tests skipped for ROCM** 
 114 out of 593

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -12,55 +12,46 @@ def show_node_info() {
...
+                show_node_info()
+                script {
+                    sh "docker build  -f Dockerfile_rocm_ci -t ${DOCKER_IMAGE}  ${DOCKER_BUILD_ARGS} ."
+                    }
+                }
             }
-        }
-        stage('Test') {
+
+        stage('Run Docker Container') {
```
Comment (lcskrishna) [inline Jenkinsfile]: Probably re-name it as "Run unit-tests"

**Comment by lcskrishna** [comment]
Minor comments, otherwise looks good to me.

File: Jenkinsfile (hunk 1 above)
Comment (gurpreet-dhami) [reply Jenkinsfile]: @lcskrishna  : done.

**Review by lcskrishna — APPROVED** [review]

File: Dockerfile_rocm_ci [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    cd mamba &&\
+    git show --oneline -s &&\
+    pip install --no-build-isolation .
+
+# Clone TE repo and submodules
+WORKDIR ${STAGE_DIR}
+ENV NVTE_FRAMEWORK=pytorch
+ENV PYTORCH_ROCM_ARCH=${PYTORCH_ROCM_ARCH_OVERRIDE}
+ENV NVTE_USE_HIPBLASLT=1
+RUN git clone --recursive https://github.com/ROCmSoftwarePlatform/TransformerEngine-private.git &&\
```
Comment (wenchenvincent) [inline Dockerfile_rocm_ci]: Change `TransformerEngine-private` to `TransformerEngine`.

File: Jenkinsfile [hunk 3]
```diff
@@ -12,55 +12,46 @@ def show_node_info() {
...
-    sh 'docker system prune -af --volumes || true'
-}
-
-def clean_up_docker_container() {
-    sh 'docker ps -a || true' // "|| true" suppresses errors
-    sh 'docker kill $(docker ps -q) || true'
-}
+DOCKER_IMAGE = "megatron-lm"
+CONTAINER_NAME = "megatron-lm-container"
+DOCKER_BUILD_ARGS = "--build-arg PYTORCH_ROCM_ARCH_OVERRIDE=gfx90a"
```
Comment (wenchenvincent) [inline Jenkinsfile:17]: what's our default build target? gfx90a or gfx942?

File: tests/unit_tests/dist_checkpointing/test_fp8.py [hunk 4]
```diff
@@ -18,8 +18,10 @@
 
 
 class TestFP8:
+
```
Comment (wenchenvincent) [inline tests/unit_tests/dist_checkpointing/test_fp8.py]: Remove this blank line?

File: run_unit_tests.sh [hunk 5]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+set -x
+export HIP_VISIBLE_DEVICES=0,1,2,3,4,5,6,7
+torchrun --nproc_per_node=8 -m pytest --color=yes -m "not flaky and not internal and not failing_on_rocm_mi250 and not failing_on_rocm" --csv output/test_report.csv tests/unit_tests/
```
Comment (wenchenvincent) [inline run_unit_tests.sh:5]: If a test only failed on MI250 and passed on MI300, how do we ensure that the test runs on MI300 with this filtering?

**Review by wenchenvincent — CHANGES_REQUESTED** [review]
Could you please address the comments?

File: tests/unit_tests/dist_checkpointing/test_fp8.py (hunk 4 above)
Comment (gurpreet-dhami) [reply tests/unit_tests/dist_checkpointing/test_fp8.py]: done

File: Dockerfile_rocm_ci (hunk 2 above)
Comment (gurpreet-dhami) [reply Dockerfile_rocm_ci]: done.

**Review by wenchenvincent — APPROVED** [review]

**Pull request merged**